ipv6/mask.go

```diff
@@ -0,0 +1,88 @@
+package ipv6
+
+import (
+	"fmt"
+	"net"
+)
+
+// Mask represents an IPv6 prefix mask. It has 0-128 leading 1s and then all
+// remaining bits are 0s
+type Mask Address
```

Comment:
Now that I look at this, I wonder about the wisdom of it. Doing it this way makes it part of the API and it cannot be changed after releasing v1.

I think the only reasons that I need it to be this way are the two type casts below but I'm beginning to think it isn't worth it. What do you think?

Reply:
This would fix it on the v4 side: https://github.com/go-addrs/addrs/pull/4/files

Reply:
Yeah I think that makes sense to me. It would allow us better control over the mask api